### PR TITLE
MODRS-4 - FOLIO-initiated sending items to the accession queue

### PR DESCRIPTION
**Pull request by khandramai**
[MODRS-4](https://issues.folio.org/browse/MODRS-4) - FOLIO-initiated sending items to the accession queue

## Purpose
When Kafka subscriber receives notification message from inventory-storage, it should check the new and old attributes of the item and send item to accession if the new effective location is "marked as remote"

## Approach
Kafka message listener was introduced for inventory.items topic

### TODOS and Open Questions

1. Listener expecting instanceId field in the message
2. **How to add appropriate permissions to Listener that calling /inventory?**

## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [x] Code coverage on new code is 80% or greater
   - [x] Duplications on new code is 3% or less
   - [x] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [x] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by khandramai** [comment]
This PR can be merged only after https://issues.folio.org/browse/MODINVSTOR-649 completion.

File: src/main/resources/dto/item.json [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+    },
+    "tags": {
+      "description": "arbitrary tags associated with this item",
+      "id": "tags",
+      "type": "object",
+      "$ref": "tags.schema"
+    },
+    "metadata": {
+      "type": "object",
+      "$ref": "metadata.schema",
```
Comment (siarhei-charniak) [inline src/main/resources/dto/item.json:391]: There is an existing file `src/main/resources/swagger.api/schemas/metadata.json`. Is it possible to use it here?

File: src/main/resources/dto/item.json [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        },
+        "suffix": {
+          "type": "string",
+          "description": "Effective Call Number Suffix is the suffix of the identifier assigned to an item or its holding and associated with the item.",
+          "readonly": true
+        },
+        "typeId": {
+          "type": "string",
+          "description": "Effective Call Number Type Id is the call number type id of the item, if available, otherwise that of the holding.",
+          "$ref": "uuid.json",
```
Comment (siarhei-charniak) [inline src/main/resources/dto/item.json:80]: There is an existing file `src/main/resources/swagger.api/schemas/uuid.json`. Is it possible to use it here?

**Review by siarhei-charniak — APPROVED** [review]
@khandramai, LGTM. Please take a look at inline comments and exclude `@Data`-annotated files from SonarCloud analysis to increase code coverage.

File: src/main/java/org/folio/rs/domain/dto/DomainEventType.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package org.folio.rs.domain.dto;
+
+public enum DomainEventType {
```
Comment (MikhailFokanov) [inline src/main/java/org/folio/rs/domain/dto/DomainEventType.java:3]: There is another type of event. If you don't declare it here, there will be an exception

**Review by MikhailFokanov — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/rs/client/EnrichHeadersClient.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.folio.spring.FolioExecutionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Log4j2
+public class EnrichHeadersClient extends Client.Default {
```
Comment (MikhailFokanov) [inline src/main/java/org/folio/rs/client/EnrichHeadersClient.java:21]: This will be removed in the next PRs, as there is an alternative in spring-folio

**Pull request merged**